fvm/evm/emulator/emulator.go

```diff
@@ -234,7 +234,17 @@ func (bl *BlockView) DryRunTransaction(
 	msg.SkipAccountChecks = true
 
 	// return without commiting the state
-	return proc.run(msg, tx.Hash(), 0, tx.Type())
+	txResult, err := proc.run(msg, tx.Hash(), 0, tx.Type())
+	if txResult.VMError == nil && txResult.ValidationError == nil {
+		// Adding `gethParams.SstoreSentryGasEIP2200` is needed for this condition:
+		// https://github.com/onflow/go-ethereum/blob/master/core/vm/operations_acl.go#L29-L32
+		txResult.GasConsumed += gethParams.SstoreSentryGasEIP2200
+		// Take into account any gas refunds, which are calculated only after
+		// transaction execution.
+		txResult.GasConsumed += txResult.GasRefund
```

Comment:
I would move this to the `GasRefund()` method I mentioned bellow

Reply:
That would break out the current functionality. We need to track `GasConsumed` and `GasRefund` separately, and only add them up for `EVM.dryRun`, which is a special case for proper gas estimation.